src/data/markdown/docs/02 javascript api/04 k6-data/1-SharedArray.md

```diff
@@ -3,26 +3,49 @@ title: SharedArray
 excerpt: 'SharedArray is an array-like object that shares the underlying memory between VUs.'
 ---
 
-`SharedArray` is an array-like object that shares the underlying memory between VUs. Its constructor takes a name for the `SharedArray` and a function which needs to return an array object itself. The function will be executed only once and its result will then be saved in memory once and copies of the elements will be given when requested. The name is needed as VUs are completely separate JS VMs and k6 needs some way to identify the `SharedArray`s that it needs to return.
+`SharedArray` is an array-like object that shares the underlying memory between VUs.
+The function executes only once, and its result is saved in memory once.
+When a script requests an element, k6 gives a _copy_ of that element.
 
-This does mean that you can have multiple such ones and even only load some of them for given VUs, although that is unlikely to have any performance benefit.
+You must construct a `SharedArray` in `init` context.
```

Comment:
```suggestion
You must construct a `SharedArray` in the [`init` context](/using-k6/test-life-cycle).
```

Not sure if that is the correct way to add the link to https://k6.io/docs/using-k6/test-life-cycle though